libheif/context.cc

```diff
@@ -2465,6 +2465,7 @@ Error HeifContext::encode_image_as_hevc(const std::shared_ptr<HeifPixelImage>& i
     if (!src_image) {
       return Error(heif_error_Unsupported_feature, heif_suberror_Unsupported_color_conversion);
     }
+    *target_nclx_profile = *src_image->get_color_profile_nclx();
```

Comment:
I don't think this is the correct way to fix this. We have an NCLX of the input image and a desired output NCLX from the `heif_encoding_options`. If both differ, the input will be transformed to match the output options.
Here, you are overwriting the target NCLX (the one that will be written to the file) with the input NCLX even if it was converted to another one (specified in the encoding options).

Reply:
IIUC, this only uses the NCLX parameters of the input if some of the output NCLX parameters are unspecified, otherwise it would still preserve the output NCLX parameters as-is.

So, for example, when saving with `options.output_nclx_profile == nullptr` (default) it would both remap these unspecified values:
https://github.com/strukturag/libheif/blob/673fb03227790cb5184c9c9aaa5c40775cb3afbf/libheif/nclx.cc#L315-L316

To the ones specified within the input NCLX profile here:
https://github.com/strukturag/libheif/blob/673fb03227790cb5184c9c9aaa5c40775cb3afbf/libheif/color-conversion/colorconversion.cc#L544-L560

This updated output NCLX profile would be propagated to the result of `convert_colorspace()` here:
https://github.com/strukturag/libheif/blob/673fb03227790cb5184c9c9aaa5c40775cb3afbf/libheif/color-conversion/colorconversion.cc#L447-L450

Reply:
You're right; this fix was incorrect.